omod/src/main/java/org/openmrs/module/htmlformentryui/fragment/controller/htmlform/EnterHtmlFormFragmentController.java

```diff
@@ -144,10 +146,27 @@ public void controller(FragmentConfiguration config, UiSessionContext sessionCon
 			        !automaticValidation);
 		}
 		
+		String visitStartDatetime = null;
+		String visitStopDatetime = null;
+		
+		//If GP timezone is true, it will convert the visitStartDatetime and visitStopDatetime to UTC and format RFC3339
+		if (visit != null) {
+			if (visit.getStartDatetime() != null) {
+				visitStartDatetime = ui.convertTimezones() ? ui.format(visit.getStartDatetime())
+				        : ui.dateToISOString(visit.getStartDatetime());
+			}
+			if (visit.getStopDatetime() != null) {
+				visitStopDatetime = ui.convertTimezones() ? ui.format(visit.getStopDatetime())
+				        : ui.dateToISOString(visit.getStopDatetime());
+			} else {
+				visitStopDatetime = ui.convertTimezones() ? ui.format(new Date()) : ui.dateToISOString(new Date());
```

Comment:
Why are we providing a visit stop date time when it is null?

Reply:
Basically that visitStopDatetime will be used on the <encounterDate> to set the max date that the user can choose from.  If the visit stop date is null, it will be `new Date()` because he cannot choose a date from the future. Maybe we could rename it for something more suitable.

Reply:
We did not do anything with it before this pull request. Why now? How is it related to timezone support?

Reply:
Lets imagine this scenario, the timzone GP is true:
The server is on UTC but the client is on UTC+12, so it is possible that the client is on a different day than the server.
On the `<encounterDate>` it should be possible for the user to select the present day, without timezone conversion the stopDatetime for `<encounterDate>`  would be the server date, because it cannot be in the future, now with timezone support, it could be on the present day on the client timezone. Basically, we are using timezone to get the right start/stop date for `<encounterDate>`